File: solutions/2789. Largest Element in an Array after Merge Operations/2789.py
class Solution:
  def maxArrayValue(self, nums: list[int]) -> int:
    ans = nums[-1]

    for i in range(len(nums) - 2, -1, -1):
      if nums[i] > ans:
        ans = nums[i]
      else:
        ans += nums[i]

    return ans

File: solutions/2789. Largest Element in an Array after Merge Operations/2789.java
class Solution {
  public long maxArrayValue(int[] nums) {
    long ans = nums[nums.length - 1];

    for (int i = nums.length - 2; i >= 0; --i)
      if (nums[i] > ans)
        ans = nums[i];
      else
        ans += nums[i];

    return ans;
  }
}
